{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import urllib.request\n",
    "import urllib.parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "url = 'http://www.rcsb.org/pdb/rest/search'\n",
    "\n",
    "data = urllib.parse.urlencode(\n",
    "    {'queryType' : 'org.pdb.query.simple.ExpTypeQuery',\n",
    "     'description' : 'Experimental Method Search: Experimental Method=SOLID-STATE NMR',\n",
    "     'mvStructure.expMethod.value' : 'SOLID-STATE NMR'}\n",
    ")\n",
    "data = data.encode('utf-8')\n",
    "headers = {\"Content-Type\":\" application/xml;charset=utf-8\"}\n",
    "\n",
    "# queryText = \"\"\"\n",
    "# <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n",
    "# <orgPdbQuery>\n",
    "# <version>B0907</version>\n",
    "# <queryType>org.pdb.query.simple.ExpTypeQuery</queryType>\n",
    "# <description>Experimental Method Search: Experimental Method=SOLID-STATE NMR</description>\n",
    "# <mvStructure.expMethod.value>SOLID-STATE NMR</mvStructure.expMethod.value>\n",
    "# </orgPdbQuery>\n",
    "# \"\"\"\n",
    "\n",
    "\n",
    "#print(\"query:\\n\", queryText)\n",
    "#print(\"querying PDB...\\n\")\n",
    "\n",
    "#data = urllib.parse.urlencode(queryText)\n",
    "#print(data)\n",
    "\n",
    "req = urllib.request.Request(url, data, headers)\n",
    "f = urllib.request.urlopen(req)\n",
    "\n",
    "result = f.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "b'\\n\\n\\n\\n\\n\\n<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\\n<html xmlns=\"http://www.w3.org/1999/xhtml\">\\n<head>\\n    <link rel=\"alternate\" type=\"application/rss+xml\" title=\"RCSB PDB - Latest Released Structures\" href=\"/pdb/rss/LastLoad\"/>\\n    <meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\"/>\\n    <title>RCSB PDB - REST Web Service \\n\\n\\n\\n\\n\\n</title>\\n    <meta http-equiv=\"Cache-Control\" content=\"max-age=3600\"/>\\n<meta http-equiv=\"Cache-Control\" content=\"private\"/>\\n<meta http-equiv=\"Cache-Control\" content=\"must-revalidate\"/>\\n<meta http-equiv=\"Pragma\" content=\"private\"/>\\n\\n<meta name=\"description\" content=\"\\n\\t\\t\\tThe PDB archive contains information about experimentally-determined structures of proteins,\\n\\t\\t\\tnucleic acids, and complex assemblies. As a member of the wwPDB, the RCSB PDB curates and\\n\\t\\t\\tannotates PDB data according to agreed upon standards. The RCSB PDB also provides a variety\\n\\t\\t\\tof tools and resources. Users can perform simple and advanced searches based on annotations\\n\\t\\t\\trelating to sequence, structure and function. These molecules are visualized, downloaded, and\\n\\t\\t\\tanalyzed by users who range from students to specialized scientists.\\n\\t\\t\\t\"/>\\n\\n<link rel=\"image_src\" href=\"http://www.rcsb.org/pdb/skins/web20/default/images/pdblogo.gif\"/>\\n\\n<meta name=\"author\" content=\"RCSB Protein Data Bank\"/>\\n<meta name=\"citation_author\" content=\"RCSB Protein Data Bank\"/>\\n\\n<meta name=\"email\" content=\"info@rcsb.org\"/>\\n<meta name=\"Charset\" content=\"UTF-8\"/>\\n<meta name=\"Distribution\" content=\"Global\"/>\\n<meta name=\"Rating\" content=\"General\"/>\\n<meta http-equiv=\"content-language\" content=\"en-US\"/>\\n\\n    <link rel=\"shortcut icon\" href=\"/pdb/img/pdb.ico\" type=\"image/x-icon\"/>\\n    <link rel=\"search\" type=\"application/opensearchdescription+xml\" title=\"RCSB Full Text or PDB ID Search \" href=\"http://www.rcsb.org/pdb/rcsb_pdb_keywordIDSearch.xml\"/>\\n    <link rel=\"search\" type=\"application/opensearchdescription+xml\" title=\"RCSB Author Search\" href=\"http://www.rcsb.org/pdb/rcsb_pdb_authorSearch.xml\"/>\\n    <link rel=\"canonical\" href=\"http://www.rcsb.org/pdb/software/rest.do\"/>\\n\\n    \\n    <meta name=\"google-site-verification\" content=\"sESjmGkFOONWjl0ez3izG4EHrjcCxzajqjlOT7UsHMo\"/>\\n    \\n    <meta name=\"google-site-verification\" content=\"OcOa2oOfcTASODQ3irE8zVJieR8Zbj0LL0xvwJrb0vE\"/>\\n    <!-- Apple Smart App Header -->\\n    <meta name=\"apple-itunes-app\" content=\"app-id=529153183\"/>\\n    \\n    <link rel=\"stylesheet\" type=\"text/css\" href=\"/pdb/assets/wro/global.css?ver=20160607-1155\"/>\\n    <script type=\"text/javascript\" src=\"/pdb/assets/wro/global.js?ver=20160607-1155\"></script>\\n    \\n    <script type=\"text/javascript\" language=\"JavaScript\">\\n        var jmolPath_global = \"/pdb/Viewers/jmol-14.4.4\";\\n        var jmolVersionNumber_global = \"14.4.4\";\\n        function getJmolGlobalPath() {\\n            alert(\"ing getJmolGlobalPath() \" + jmolPath_global);\\n            return jmolPath_global;\\n        }\\n    </script>\\n    \\n</head>\\n<body>\\n<div id=\"helpOverlay\" class=\"webDisplay\" onclick=\"$(\\'#helpOverlay\\').hide();$(\\'#helpContent\\').hide();showApplets();\"></div>\\n<div id=\"helpContent\" class=\"webDisplay\">\\n    \\n\\n\\n\\n\\n\\t<div class=\"se_spacer\">&nbsp;</div>\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"lb_head3\" style=\"padding-top:3px;\">\\n\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n    <iframe id=\"helpFrame\" name=\"helpFrame\" src=\"/pdb/skins/web20/templates/blank.html\" width=\"100%\" height=\"525px\"></iframe>\\n    \\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n    <div style=\"float:right;\">\\n        <a id=\"helpPopout\" href=\"\" style=\"color: #ffffff;\" onclick=\"$(\\'#helpOverlay\\').hide();$(\\'#helpContent\\').hide();showApplets();window.open($(\\'#helpPopout\\').attr(\\'href\\'),\\'helpWindow\\',\\'menubar=no,width=880,height=550,toolbar=yes,scrollbars=yes,resizable=no\\').focus();return false;\" target=\"helpWindow\">POP-OUT</a> |\\n        <a style=\"color: #ffffff;\" onclick=\"$(\\'#helpFrame\\').attr(\\'src\\',\\'\\');$(\\'#helpOverlay\\').hide();$(\\'#helpContent\\').hide();showApplets();return false;\">CLOSE</a>\\n    </div>\\n</div>\\n<table class=\"perc100\" id=\"top\">\\n    <tr>\\n        <td style=\"padding:0;\">\\n            <!-- Navigation Menu -->\\n            <div id=\"horizontal_navigation\">\\n                \\n<ul id=\"nav\" class=\"container\">\\n    <li class=\"hometab\">\\n        <a href=\"/pdb/home/home.do\">RCSB PDB</a>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Deposit <b class=\"caret\"></b></a>\\n        <div class=\"dropdown-menu multi-column\" id=\"DepositDropdownNav\">\\n            <ul class=\"column-left\">\\n                <!-- Prepare Data -->\\n                <li><h4>Prepare Data</h4></li>\\n                <li><a href=\"http://pdb-extract.wwpdb.org/\" target=\"_blank\">pdb_extract</a></li>\\n                <li><a href=\"http://sf-tool.wwpdb.org/\" target=\"_blank\">SF-Tool</a></li>\\n                <li><a href=\"http://ligand-expo.rcsb.org/\" target=\"_blank\">Ligand Expo</a></li>\\n                <li><a href=\"http://sw-tools.pdb.org/apps/MAXIT/index.html\" target=\"_blank\">MAXIT</a></li>\\n                <hr>\\n                <!-- Validate Data -->\\n                <li><h4>Validate Data</h4></li>\\n                <li>\\n                    <a href=\"http://www.wwpdb.org/validation/validation-servers.php\" target=\"_blank\">Validation Server</a>\\n                </li>\\n                <li>\\n                    <a href=\"http://www.wwpdb.org/documentation/journals.php\" target=\"_blank\">Information for Journals</a>\\n                </li>\\n                <li><a href=\"http://www.wwpdb.org/task/validation-task-forces.php\" target=\"_blank\">Validation Task Forces</a>\\n                </li>\\n            </ul>\\n            <ul class=\"column-right\">\\n                <!-- Deposit Data -->\\n                <li><h4>Deposit Data</h4></li>\\n                <li><h5>wwPDB Deposition & Annotation System</h5></li>\\n                <li><a href=\"http://deposit.wwpdb.org/deposition/\" target=\"_blank\">All Methods</a></li>\\n                <li><h5>Legacy Deposition</h5></li>\\n                <li><a href=\"http://emdep.rutgers.edu/index.jsp\" target=\"_blank\">EmDep (EM)</a></li>\\n                <li><a href=\"http://deposit.bmrb.wisc.edu/bmrb-adit/\" target=\"_blank\">ADIT-NMR</a></li>\\n                <hr>\\n                <!-- Deposit Help & Resources -->\\n                <li><h4>Deposition Help & Resources</h4></li>\\n                <li><a href=\"http://www.wwpdb.org/deposition/faq.php\" target=\"_blank\">Deposit FAQ</a></li>\\n                <li><a href=\"http://www.wwpdb.org/deposition/tutorial.php\" target=\"_blank\">Tutorials</a></li>\\n                <li><a href=\"http://www.wwpdb.org/documentation/procedure.php\" target=\"_blank\">Annotation Policies</a></li>\\n                <li><a href=\"http://www.wwpdb.org/documentation/policy.php\" target=\"_blank\">Processing Procedures</a></li>\\n                <li><a href=\"http://mmcif.wwpdb.org/\" target=\"_blank\">PDBx/mmCIF Dictionary</a></li>\\n                <li><a href=\"http://www.wwpdb.org/documentation/file-format.php\" target=\"_blank\">PDB Format Guide</a></li>\\n\\n                <li><a href=\"http://biosync.sbkb.org/\" target=\"_blank\">BioSync Beamlines / Facilities</a></li>\\n                <li><a href=\"http://sw-tools.pdb.org/\" target=\"_blank\">Related Tools</a></li>\\n            </ul>\\n        </div>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Search <b class=\"caret\"></b></a>\\n        <ul class=\"dropdown-menu\">\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#AdvancedSearch\">Advanced Search</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#Drilldown\">Drilldown Search</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#Unreleased\">Unreleased & New Entries</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#Sequences\">Sequences</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#ligands\">Ligands</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#Drugs\">Drugs & Drug Targets</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/search.jsp#Annotation\">Browse by Annotation</a></li>\\n            <li><a href=\"/pdb/static.do?p=general_information/pdb_statistics/index.html\">PDB Statistics</a></li>\\n        </ul>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Visualize <b class=\"caret\"></b></a>\\n        <ul class=\"dropdown-menu\">\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/visualize.jsp#Jmol\">Jmol</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/visualize.jsp#RCSBviewer\">RCSB PDB Viewers</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/visualize.jsp#LigandExplorer\">Ligand Explorer</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/visualize.jsp#Poseview\">PoseView</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/visualize.jsp#ProteinFeature\">Protein Feature View</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/visualize.jsp#HumanGene\">Human Gene View</a></li>\\n        </ul>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Analyze <b class=\"caret\"></b></a>\\n        <ul class=\"dropdown-menu\">\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/analyze.jsp#Sequence\">Sequence & Structure Alignment</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/analyze.jsp#Symmetry\">Protein Symmetry</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/analyze.jsp#Quality\">Structure Quality</a></li>\\n            <li><a href=\"/pdb/chromosome.do\">Map Genomic Position to Protein</a></li>\\n            <li><a href=\"/pdb/static.do?p=general_information/web_links/index.html\">Third Party Tools</a></li>\\n        </ul>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Download <b class=\"caret\"></b></a>\\n        <ul class=\"dropdown-menu\">\\n            <li><a href=\"/pdb/download/download.do#Structures\">Structures</a></li>\\n            <li><a href=\"/pdb/download/download.do#FASTA\">Sequences</a></li>\\n            <li><a href=\"/pdb/download/download.do#Ligands\">Ligands</a></li>\\n            <li><a href=\"/pdb/static.do?p=download/ftp/index.html\">FTP Archive & Services</a></li>\\n            <li><a href=\"/pdb/static.do?p=download/http/index.html\">HTTP Services</a></li>\\n            <li><a href=\"/pdb/secondary.do?p=v2/secondary/other_download.jsp\">Other Services</a></li>\\n            <li><a href=\"/pdb/software/rest.do\">RESTful Web Services</a></li>\\n        </ul>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Learn <b class=\"caret\"></b></a>\\n        <ul class=\"dropdown-menu\">\\n            <li><h4>\\n                <a href=\"http://pdb101.rcsb.org\" target=\"_blank\"><img src=\"/pdb/v2/common/images/pdb101Logo.png\" class=\"image_submenu\"></a>\\n            </h4></li>\\n            <li><a href=\"http://pdb101.rcsb.org/more/about-pdb-101\" target=\"_blank\">About</a></li>\\n            <li><a href=\"http://pdb101.rcsb.org/motm/motm-by-title\" target=\"_blank\">Molecule of the Month</a></li>\\n            <li><a href=\"http://pdb101.rcsb.org/browse\" target=\"_blank\">Browse all Resources</a></li>\\n            <li><a href=\"http://pdb101.rcsb.org/learn/resources/paper_models\" target=\"_blank\">Educational Resources</a></li>\\n            <li><a href=\"http://pdb101.rcsb.org/teach/overview\" target=\"_blank\">Curricula</a></li>\\n            <li><a href=\"http://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data\" target=\"_blank\">Guide to PDB Data</a></li>\\n            <li><a href=\"http://pdb101.rcsb.org/news\" target=\"_blank\">News and Events</a></li>\\n        </ul>\\n    </li>\\n    <li class=\"dropdown\">\\n        <a href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">More <b class=\"caret\"></b></a>\\n        <ul class=\"dropdown-menu\">\\n            <li><a href=\"/pdb/home/contactUs.do\">Contact Us</a></li>\\n            <li>\\n                <a href=\"/pdb/static.do?p=general_information/about_pdb/policies_references.html#References\">Citing Us</a>\\n            </li>\\n            <li>\\n                <a href=\"/pdb/static.do?p=general_information/about_pdb/policies_references.html\">Usage & Privacy</a>\\n            </li>\\n            <li><a href=\"/pdb/staticHelp.do?p=help/index.html\">Help Topics</a></li>\\n            <li><a href=\"/pdb/static.do?p=general_information/about_pdb/index.html\">About RCSB PDB</a></li>\\n            <li><a href=\"/pdb/static.do?p=general_information/about_pdb/contact/pdb_members.html\">Team Members</a>\\n            </li>\\n            <li><a href=\"/pdb/static.do?p=general_information/about_pdb/pdbac.html\">Advisory Committee</a></li>\\n            <li><a href=\"/pdb/static.do?p=general_information/news_publications/index.html\">News</a></li>\\n            <li>\\n                <a href=\"/pdb/static.do?p=general_information/news_publications/articles/index.html\">Publications</a>\\n            </li>\\n            <li>\\n                <a href=\"/pdb/static.do?p=mobile/RCSBapp.html\">RCSB PDB <i>Mobile</i></a>\\n            </li>\\n            <li><a href=\"/pdb/static.do?p=general_information/about_pdb/contact/job_listings.html\">Careers</a></li>\\n        </ul>\\n    </li>\\n    <!-- MyPDB Login -->    <!-- Show this block: if @ \"null\" state, user has not logged into account. Hover over. -->\\n    <li class=\"dropdown dropdown_account\" id=\"mypdb_loginMenu\">\\n        <button href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">MyPDB Login</button>\\n        <div id=\"dropdown-mypdb_form\">\\n            <form method=\"post\" action=\"/pdb/admin/Logon.do\" onsubmit=\"return true;\" name=\"Logon\" id=\"lhm_mypdb_Logon\">\\n                <label for=\"lhm_mypdb_username\">Username</label>\\n                <input type=\"text\" id=\"lhm_mypdb_username\" value=\"\" size=\"10\" name=\"username\"/>\\n                <label for=\"lhm_mypdb_password\">Password</label>\\n                <input type=\"password\" id=\"lhm_mypdb_password\" size=\"10\" name=\"password\"/>\\n                <input type=\"hidden\" id=\"lhm_mypdb_previousRequestUrl\" value=\"\" name=\"previousRequestUrl\"/>\\n                <button type=\"submit\" value=\"Login\" name=\"method\"/>\\n                Login</button>\\n                <div class=\"password_account\">\\n                    <a href=\"/pdb/mypdb/forgotPassword.do\">Forget your Password?</a></br>\\n                    <a href=\"/pdb/mypdb/signup.do\">Register a New Account</a>\\n                </div>\\n                <div class=\"clear\"></div>\\n            </form>\\n        </div>\\n    </li>\\n    <!-- Show this block: if @ \"login\" state, user has submitted account information. -->\\n    <li class=\"dropdown dropdown_account hide\" id=\"mypdb_userMenu\">\\n        <button href=\"#\" class=\"dropdown-toggle\" data-toggle=\"dropdown\">Welcome <span id=\"mypdb_userVal\"></span>!\\n            <i class=\"fa fa-cog fa-2\"></i></button>\\n        <ul class=\"dropdown-menu\" id=\"dropdown-mypdb_options\">\\n            <li><a href=\"/pdb/mypdb/queryManager.do\" title=\"Show MyPDB\">Saved Query Manager</a></li>\\n            <li><a href=\"/pdb/mypdb/mobileAnnotations.do\">Personal Annotations</a></li>\\n            <li><a href=\"/pdb/mypdb/updateUser.do\" title=\"User Account Details\">User Account</a></li>\\n            <div id=\"mypdb_adminMenu\">\\n                <li class=\"divider\"></li>\\n                <li><a href=\"/pdb/mypdb/showUsers.do\">Users</a></li>\\n                <li><a href=\"/pdb/mypdb/myPdbGroupMgr.do\">Groups</a></li>\\n                <li><a href=\"/pdb/admin/viewStatus.do\">Status</a></li>\\n                <li><a href=\"/pdb/admin/PropertiesView.do\">Properties</a></li>\\n                <li><a href=\"/pdb/admin/viewSessions.do\">Sessions</a></li>\\n                <li><a href=\"/pdb/admin/LoadersView.do\">Loaders</a></li>\\n                <li><a href=\"/pdb/admin/runUpdate.do\">Update</a></li>\\n                <li><a href=\"/pdb/admin/viewLoadResults.do\">Load Results</a></li>\\n                <li><a href=\"/pdb/admin/runUtils.do\">Structure Utilities</a></li>\\n                <li><a href=\"/pdb/admin/reconcile.do\">Reconcile</a></li>\\n            </div>\\n            <li class=\"divider\"></li>\\n            <li><a id=\"lhm_mypdb_logoff\" href=\"/pdb/Logoff.do\" title=\"Logout\">Logout</a></li>\\n        </ul>\\n    </li>\\n    <!-- -->\\n    <div class=\"clear\"></div>\\n</ul>\\n<div class=\"clear\"></div>\\n            </div>\\n            <!-- Hybrid version of header -->\\n            \\n\\n\\n\\n\\n\\n\\n\\n\\n\\n<div id=\"header_RCSB\">\\n    <div class=\"container\">\\n        <div class=\"row\">\\n            <div class=\"col-md-5\">\\n                <a href=\"/pdb/home/home.do\"><img src=\"/pdb/v2/common/images/rcsb_logo.png\" id=\"rcsblogo\" alt=\"RCSB PDB\"></a>\\n                <h2 id=\"tagline\">An Information Portal to <a id=\"CurrentStructureCount\" href=\"/pdb/search/smartSubquery.do?experimentalMethod=ignore&amp;smartSearchSubtype=HoldingsQuery&amp;moleculeType=ignore\" class=\"emphasis\">119480</a> Biological Macromolecular Structures</h2>\\n                <div class=\"clearfix\"></div>\\n            </div>\\n\\n            <!-- Search Bar -->\\n            <div class=\"col-md-7\">\\n                <form action=\"/pdb/search/navbarsearch.do\" method=\"get\" name=\"headerQueryForm\" autocomplete=\"off\"\\n                      role=\"search\" class=\"form-inline\" id=\"headerQueryForm\">\\n\\n                    <input type=\"hidden\" name=\"f\" id=\"fixeditem\" />\\n                    <!-- Input Field -->\\n                    <input type=\"text\" class=\"form-control\" name=\"q\" id=\"autosearch_SearchBar\" autocomplete=\"off\"\\n                           onfocus=\"if(this.value == \\'Search by PDB ID, author, macromolecule, sequence, or ligands\\') { this.value = \\'\\'; }\"\\n                           onblur=\"if(this.value == \\'\\') { this.value = \\'Search by PDB ID, author, macromolecule, sequence, or ligands\\'; }\"\\n                           value=\"Search by PDB ID, author, macromolecule, sequence, or ligands\"/>\\n                    <!-- Add in a clear button inside my HTML text input box like iPhone -->\\n                    <!-- Submission -->\\n                    <button type=\"submit\" class=\"btn_search btn-default\" id=\"searchbutton\">Go</button>\\n                </form>\\n\\n                <span class=\"quicklinks\">\\n                    <a href=\"/pdb/search/advSearch.do?search=new\">Advanced<span class=\"hidden-xs\"> Search</span></a> |\\n                    <a href=\"/pdb/browse/jbrowse.do?t=4&amp;useMenu=no\">Browse <span class=\"hidden-xs\">by Annotations</span></a>\\n                    <span id=\"SearchBarHistory\" class=\"hide\"> | <a href=\"/pdb/queries/queries.do\"><span class=\"hidden-xs\">Search </span>History<span id=\"SearchBarHistoryCount\" class=\"queryCountValSpan\"></span></a></span>\\n                    <span id=\"SearchBarPreviousResults\" class=\"hide\"> | <a href=\"/pdb/results/results.do?startAt=0\"><span class=\"hidden-xs\">Previous </span>Results<span id=\"SearchBarPreviousResultsCount\" class=\"queryResultCountValSpan\"></span></a></span>\\n                </span>\\n            </div>\\n        </div>\\n        <ul id=\"PDBmembers\">\\n            <li><a href=\"/pdb/101/structural_view_of_biology.do\" target=\"_blank\"><img src=\"/pdb/v2/common/images/pdb101Logo.png\" alt=\"PDB-101\" id=\"pdb101logo\"></a></li>\\n            <li><a href=\"http://www.wwpdb.org/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_wwpdb.png\" alt=\"wwPDB\" id=\"wwpdblogo\"></a></li>\\n            <li><a href=\"http://www.emdatabank.org/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_EMDatabank.png\" alt=\"EMDataBank\" id=\"emdatabanklogo\"></a></li>\\n            <li><a href=\"http://ndbserver.rutgers.edu/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_NDB.png\" alt=\"NDB Nucleic Acid Database\" id=\"ndblogo\"></a></li>\\n            <li><a href=\"http://sbkb.org/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_SBKB.png\" alt=\"SBKB Structural Biology Knowledgebase\" id=\"sbkblogo\"></a></li>\\n            <li><a href=\"http://foundation.wwpdb.org/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_wwPDBFoundation.png\" alt=\"wwPDB Foundation\" id=\"wwpdbfoundationlogo\"></a></li>\\n        </ul>\\n        <ul class=\"fa-ul\" id=\"socialmedia_section\">\\n            <li><a href=\"https://www.facebook.com/RCSBPDB\" target=\"_blank\"><i class=\"fa fa-facebook-square fa-lg\"></i></a></li>\\n            <li><a href=\"https://twitter.com/buildmodels\" target=\"_blank\"><i class=\"fa fa-twitter fa-lg\"></i></a></li>\\n            <li><a href=\"https://www.youtube.com/user/RCSBProteinDataBank\" target=\"_blank\"><i class=\"fa fa-youtube-play fa-lg\"></i></a></li>\\n            <li><a href=\"https://itunes.apple.com/us/app/rcsb-protein-data-bank/id529153183?ls=1&mt=8\" target=\"_blank\"><i class=\"fa fa-apple fa-lg\"></i></a></li>\\n            <li><a href=\"https://play.google.com/store/apps/details?id=org.rcsb.rcsbpdbapp\" target=\"_blank\"><i class=\"fa fa-android fa-lg\"></i></a></li>\\n            <li><a href=\"https://github.com/rcsb\" target=\"_blank\"><i class=\"fa fa-github fa-lg\"></i></a></li>\\n        </ul>\\n    </div>\\n    <div class=\"clear\"></div>\\n</div>\\n<div class=\"clear\"></div>\\n\\n            <div id=\"mainWrapper\" class=\"container\">\\n                <table class=\"wrapper\" style=\"width: 100%;\">\\n                    <tr>\\n                        <td class=\"mainContent\">\\n                            <div id=\"content\">\\n                                <noscript>\\n                                    <div style=\"padding-left:8px;font-weight:bold;color:#FF0000;font-size:16px\">\\n                                        This browser is either not Javascript enabled or has it turned off.<br/> This site will not function correctly without Javascript.\\n                                    </div>\\n                                </noscript>\\n                                \\n                                   \\n                                \\n\\n\\n\\n\\n\\n\\n\\n<h1>The RCSB PDB RESTful Web Service interface</h1>\\nThe RCSB PDB supports RESTful (REpresentational State Transfer) Web Services to make accessing data easier. \\nPlease use these services instead of screen-scraping. \\n\\n\\n<p>\\nGenerally we are trying to implement two types of services for our RESTful interface:\\n</p>\\n<ul>\\n<li><b>Search services</b>: to return a list of IDs (e.g., PDB IDs, chain IDs, ligand IDs)</li>\\n<li><b>Fetch services</b>: to return data given a ID (e.g. reports, descriptions, data items)</li>\\n</ul>\\n\\n<hr/>\\n\\nThe services below are currently provided; please <a href=\"/pdb/home/contactUs.do\">let us know</a> if you have additional suggestions.\\n\\n<h3>SEARCH services</h3>\\n\\n<ul>\\n\\t<li>A generic <a href=\"#search\">SEARCH service</a> allowing to POST advanced queries</li>\\t\\n\\t<li>Search for ligands and PDB IDs based on a <a href=\"#smiles\">SMILES query</a></li>\\t\\n</ul>\\n\\n<div class=\"se_spacer\" style=\"clear:both;\">&nbsp;</div>\\n\\n<h3>FETCH services</h3>\\n\\n\\n<h4>Custom Reports</h4>\\n<ul>\\n\\t<li>Retrieve a <a href=\"/pdb/software/wsreport.do\">custom report with a list of PDB IDs and fields</a></li>\\n</ul>\\n<div class=\"se_spacer\" style=\"clear:both;\">&nbsp;</div>\\n\\n<h4>Descriptions</h4>\\n<ul>\\t\\n\\t<li>Get <a href=\"#descPDB\">descriptions for the whole PDB file</a></li>\\n\\t<li>Get <a href=\"#descEntity\">descriptions for PDB entities</a></li>\\n\\t<li>Get <a href=\"#descComp\">descriptions of chemical components</a></li>\\n\\t<li>Get <a href=\"#ligands\">ligands</a> that are present in a single PDB entry</li>\\t\\n</ul>\\n<div class=\"se_spacer\" style=\"clear:both;\">&nbsp;</div>\\n<h4>Lists and status</h4>\\n<ul>\\n\\t<li>Get a PDB ID\\'s <a href=\"#releaseStatus\">release status</a></li>\\n\\t<li>Get a list of all <a href=\"#current\">currently released PDB IDs</a></li>\\t\\n\\t<li>Get a list of <a href=\"#unreleased\">unreleased PDB IDs</a></li>\\t\\n\\t<li>Get the <a href=\"#statusSequence\">pre-release sequences in FASTA format</a></li>\\n\\t\\t\\t\\n</ul>\\n<div class=\"se_spacer\" style=\"clear:both;\">&nbsp;</div>\\n<h4>Annotations</h4>\\t\\t\\n<ul>\\n\\t<li><a href=\"#dasfeatures\">Third-party annotations mapped onto PDB chains</a></li>\\n\\t<li>A <a href=\"#dasalignment\">PDB to UniProtKB mappings</a></li>\\n\\t<li><a href=\"#pfam\">Pfam mappings</a> for PDB</li>\\n\\t<li>Access <a href=\"#go\">Gene Ontology terms</a> for a PDB chain.</li>\\n</ul>\\n<div class=\"se_spacer\" style=\"clear:both;\">&nbsp;</div>\\n<h4>Sequence and Structure Clusters</h4>\\n<ul>\\t\\n\\t<li><a href=\"#seqClust\">Sequence and Structure cluster</a> related services</li>\\n</ul>\\n\\n<div class=\"se_spacer\" style=\"clear:both;\">&nbsp;</div>\\n\\n<hr/>\\n\\n<hr/>\\n\\n<a name=\"descPDB\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>PDB File Description Service</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nAccess general information about a PDB entry (e.g., Header information).\\n<ul>\\n<li>Example: Access the information for one file: <a href=\"/pdb/rest/describePDB?structureId=4hhb\">/pdb/rest/describePDB?structureId=4hhb</a></li>\\n\\n<li>Example: Access the information only for multiple files: <a href=\"/pdb/rest/describePDB?structureId=4hhb,1hhb\">/pdb/rest/describePDB?structureId=4hhb,1hhb</a></li>\\n</ul>\\n\\nDescribe the molecular entities that are contained in a PDB entry. Also provides information on the number of available biological assemblies.\\n\\n<ul>\\n\\t<li>Example: Access for one PDB ID<a href=\"/pdb/rest/getEntityInfo?structureId=1hv4\">/pdb/rest/getEntityInfo?structureId=1hv4</a></li>\\n\\t<li>Example: Access the information for all PDB IDs: <a href=\"/pdb/rest/getEntityInfo\">/pdb/rest/getEntityInfo</a></li>\\t\\n\\t\\t\\n</ul>\\n\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"descEntity\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Descriptions of PDB entities</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nAccess the descriptions of the entities that are contained in a PDB file. \\n<ul>\\n<li>Example: Access the information for all entities in the file: <a href=\"/pdb/rest/describeMol?structureId=4hhb\">/pdb/rest/describeMol?structureId=4hhb</a></li>\\n\\n<li>Example: Access the information only for a single PDB chain: <a href=\"/pdb/rest/describeMol?structureId=4hhb.A\">/pdb/rest/describeMol?structureId=4hhb.A</a></li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n<a name=\"descComp\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Descriptions of Chemical Components</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nAccess the descriptions of the chemical components based on their IDs. XML contains a PDB ID as an example of where it can be found.\\n<ul>\\n<li>Example: Access the information for one components: <a href=\"/pdb/rest/describeHet?chemicalID=NAG\">/pdb/rest/describeHet?chemicalID=NAG</a></li>\\n\\n<li>Example: Access the information for multiple components: <a href=\"/pdb/rest/describeHet?chemicalID=NAG,EBW\">/pdb/rest/describeHet?chemicalID=NAG,EBW</a></li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n<a name=\"current\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>List all current/obsolete PDB IDs</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nGet a list of all current PDB IDs.\\n<ul>\\n\\t<li>Example: List all current PDB IDs: <a href=\"/pdb/rest/getCurrent\">/pdb/rest/getCurrent</a></li>\\t\\n</ul>\\nGet a list of all obsolete PDB IDs.\\n<ul>\\n\\t<li>Example: List all obsolete PDB IDs: <a href=\"/pdb/rest/getObsolete\">/pdb/rest/getObsolete</a></li>\\t\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"blastPdb\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>BLAST search on the PDB database</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nBLAST with Sequence.\\n<ul>\\n\\t<li>Example:  HTML output: <a href=/pdb/rest/getBlastPDB1?sequence=VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTAVAHVDDMPNAL&eCutOff=10.0&matrix=BLOSUM62&outputFormat=HTML>/pdb/rest/getBlastPDB1?sequence=VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTAVAHVDDMPNAL&eCutOff=10.0&matrix=BLOSUM62&outputFormat=HTML</a> </li>\\n\\t<li>Example:  XML output: <a href=/pdb/rest/getBlastPDB1?sequence=VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTAVAHVDDMPNAL&eCutOff=10.0&matrix=BLOSUM62&outputFormat=XML>/pdb/rest/getBlastPDB1?sequence=VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTAVAHVDDMPNAL&eCutOff=10.0&matrix=BLOSUM62&outputFormat=XML</a> </li>\\n\\t<li>Example:  Plain text output: <a href=/pdb/rest/getBlastPDB1?sequence=VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTAVAHVDDMPNAL&eCutOff=10.0&matrix=BLOSUM62>/pdb/rest/getBlastPDB1?sequence=VLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHGKKVADALTAVAHVDDMPNAL&eCutOff=10.0&matrix=BLOSUM62</a> </li>\\t\\n</ul>\\n\\nBLAST with Structure ID and Chain ID.\\n<ul>\\n\\t<li>Example:  HTML output: <a href=/pdb/rest/getBlastPDB2?structureId=4HHB&chainId=A&eCutOff=10.0&matrix=BLOSUM62&outputFormat=HTML>/pdb/rest/getBlastPDB2?structureId=4HHB&chainId=A&eCutOff=10.0&matrix=BLOSUM62&outputFormat=HTML</a> </li>\\n\\t<li>Example:  XML output: <a href=/pdb/rest/getBlastPDB2?structureId=4HHB&chainId=A&eCutOff=10.0&matrix=BLOSUM62&outputFormat=XML>/pdb/rest/getBlastPDB2?structureId=4HHB&chainId=A&eCutOff=10.0&matrix=BLOSUM62&outputFormat=XML</a> </li>\\n\\t<li>Example:  Plain text output: <a href=/pdb/rest/getBlastPDB2?structureId=4HHB&chainId=A&eCutOff=10.0&matrix=BLOSUM62>/pdb/rest/getBlastPDB2?structureId=4HHB&chainId=A&eCutOff=10.0&matrix=BLOSUM62</a> </li>\\t\\n</ul>\\n\\nView a Java example that submits a BLAST search via a POST:\\n\\t\\t<a href=\"static.do?p=/software/webservices/PostBLASTQuery.jsp\">PostBLASTQuery.java</a>\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n<a name=\"releaseStatus\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Information about release status</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nGet Information about the release status for one or multiple PDB ids.\\n<ul>\\n\\t<li>Example: Access the information for a one PDB ID: <a href=\"/pdb/rest/idStatus?structureId=4HHB\">/pdb/rest/idStatus?structureId=4HHB</a></li>\\n\\t<li>Example: Access the information for a set of PDB IDs: <a href=\"/pdb/rest/idStatus?structureId=450D,451D\">/pdb/rest/idStatus?structureId=450D,451D</a></li>\\n\\t\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"unreleased\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>List unreleased PDB IDs</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nGet a list of PDB IDs that have not been released yet.\\n<ul>\\n\\t<li>Example: Access the information for a subset of PDB IDs: <a href=\"/pdb/rest/getUnreleased?structureId=450D,451D\">/pdb/rest/getUnreleased?structureId=450D,451D</a></li>\\n\\t<li>Example: Access the information for all unreleased PDB IDs: <a href=\"/pdb/rest/getUnreleased\">/pdb/rest/getUnreleased</a></li>\\n\\t<li>Example: Access the information of unreleased PDB IDs based on SG Project Initials: <a href=\"/pdb/rest/getUnreleased?sgInit=SSGCID\">/pdb/rest/getUnreleased?sgInit=SSGCID</a></li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n<a name=\"statusSequence\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>List pre-release sequences in FASTA format</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nGet a list of pre-release sequence in FASTA format.\\n<ul>\\n\\t<li>Example: Access the information for a subset of PDB IDs: <a href=\"/pdb/rest/getStatusSequence?structureId=3MU6,3QV1,3SSF\">/pdb/rest/getStatusSequence?structureId=3MU6,3QV1,3SSF</a></li>\\n\\t<li>Example: Access the information for a subset of PDB IDs, entity type, and range of deposition date: <a href=\"/pdb/rest/getStatusSequence?entityType=RNA&depositionDateMin=2011-07-01&depositionDateMax=2011-07-30\">/pdb/rest/getStatusSequence?entityType=RNA&depositionDateMin=2011-07-01&depositionDateMax=2011-07-30</a>\\n\\t\\t(The entity type can be <b>RNA</b>, <b>DNA</b>, and <b>Polypeptide</b>.)\\n\\t</li>\\n\\t<li>Example: Access the information for all unreleased PDB IDs: <a href=\"/pdb/rest/getStatusSequence\">/pdb/rest/getStatusSequence</a></li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"ligands\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Ligands in PDB files</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nList the ligands that can be found in a PDB entry.\\n<ul>\\n<li>Example: <a href=\"/pdb/rest/ligandInfo?structureId=4HHB\">/pdb/rest/ligandInfo?structureId=4HHB</a></li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"go\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Gene Ontology terms for PDB chains</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\nList the <b>Gene Ontology terms</b> that have been annotated for a PDB chain. (Same as on the Annotations tab in the structure explorer pages).\\n\\n<ul>\\n<li>Example: <a href=\"/pdb/rest/goTerms?structureId=4HHB\">/pdb/rest/goTerms?structureId=4HHB</a></li>\\n<li>Example: <a href=\"/pdb/rest/goTerms?structureId=4HHB.A\">/pdb/rest/goTerms?structureId=4HHB.A</a></li>\\n<li>Example: <a href=\"/pdb/rest/goTerms?structureId=4hHHB.A,1A4W.H\">/pdb/rest/goTerms?structureId=4HHB.A,1A4W.H</a></li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\t\\n\\n\\n\\n<a name=\"pfam\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Pfam annotations for PDB entries</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n\\nThe <b>Pfam annotations</b> that are displayed on the RCSB PDB web site are calculated by weekly scripting the \\n<a href=\"http://hmmer.janelia.org/\">Hmmer web site</a>. The calculated annotations can be downloaded using any of these options:\\n<ul>\\n\\t<li>per PDB ID: <a href=\"/pdb/rest/hmmer?structureId=4hhb\">/pdb/rest/hmmer?structureId=4hhb</a></li>\\n\\t<li>all annotations in one XML file <a href=\"/pdb/rest/hmmer\">/pdb/rest/hmmer</a></li>\\n\\t<li>all annotations in one tab-delimited file <a href=\"/pdb/rest/hmmer?file=hmmer_pdb_all.txt\">/pdb/rest/hmmer?file=hmmer_pdb_all.txt</a></li>\\n\\t\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\t\\n\\n\\n\\n<a name=\"seqClust\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Sequence and Structure cluster related web services</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n\\nVarious requests related to the sequence and 3D - structure clustering are available:\\n\\n<ul>\\n<li><a href=\"/pdb/rest/sequenceCluster?cluster=40&structureId=4hhb.A\">/pdb/rest/sequenceCluster?cluster=40&structureId=4hhb.A</a> Get all the sequence cluster at 40% sequence ID for a particular chain ID.</li>\\n\\n<li><a href=\"/pdb/rest/representatives?cluster=40\">/pdb/rest/representatives?cluster=40</a> Get all representative chains from the 40% sequence clustering.</li>\\n<li><a href=\"/pdb/rest/representatives?structureId=4hhb.A\">/pdb/rest/representatives?structureId=4hhb.A</a> Get the representative for a chain (the chain on Rank 1  in the sequence clustering).</li>\\n\\n<li><a href=\"/pdb/rest/representativeDomains?cluster=40\">/pdb/rest/representativeDomains?cluster=40</a> Get all representative domains that are used for the systematic structure alignments.</li>\\n<li><a href=\"/pdb/rest/representativeDomains?structureId=4hhb.A\">/pdb/rest/representativeDomains?structureId=4hhb.A</a> Get the representatives domains for a chain.</li>\\n</ul>\\n\\n\\t<p>\\n\\t\\tThe names of the representative domains are either SCOP domain IDs (starting with &quot;d&quot;) or domain IDs as have been assigned with the Protein Domain Parser software (&quot;PDP:&quot;)\\n\\t</p>\\n\\n\\n\\t<p>\\t\\n\\tThe all vs. all structural similarity results table for a representative chain can be downloaded in XML. \\n\\t\\n\\tFor example, this returns Rank, PDB.Chain, Description,  P-value, Score, RMSD, Len1, Len2, %Sim1, \\n\\tand %Sim2 for 3BMV.A \\n\\t<a href=\"/pdb/explorer/structCompXMLData.jsp?method=pw_fatcat&chain=d3bmva1&page=1&rows=10&prettyXML\">/pdb/explorer/structCompXMLData.jsp?method=pw_fatcat&chain=d3bmva1&page=1&rows=10&prettyXML</a>.\\n\\t\\n\\tNote: A maximum 2000 rows can be returned through this URL. To fetch all approx. 17,000 results for a chain,you need to slice through the results using the <i>page</i> parameter.\\n\\n\\t<p/>\\t\\n\\tThere are additional parameters for this URL, which allow sorting. e.g. this sorts by P-value\\n\\t\\n\\t<a href=\"/pdb/explorer/structCompXMLData.jsp?method=pw_fatcat&showAllResults=false&chain=d1iarb1&rows=15&page=1&sidx=probability&sord=asc\">/pdb/explorer/structCompXMLData.jsp?method=pw_fatcat&showAllResults=false&chain=d1iarb1&rows=15&page=1&sidx=probability&sord=asc</a>\\n\\n\\tParameters that can be used for sorting (the value of the sidx parameter) are: name2,desc,probability, score, rmsdOpt, len1, len2, pid, sim1, sim2.\\n\\t</p>\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n\\n\\n<a name=\"thirdPartyAnnotations\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Third-party annotations and PDB to UniProtKB mapping</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n\\nDAS Service: <br/>\\n\\nThe RESTful services for providing third-party annotations (SCOP, CATH, Pfam, etc.) follows the DAS protocol. The DAS server is available from <a href=\"/pdb/rest/das/\">/pdb/rest/das</a>. \\nThe DAS protocol command <i>SOURCES</i> (description of the provided sources) is available from <a href=\"/pdb/rest/das/sources\">/pdb/rest/das/sources</a>.\\n\\n\\nAt the present there are two DAS sources provided:\\n<ul id=\"dasfeatures\">\\n<li>\\n\\t<p><b>pdbchainfeatures:</b></p>\\n\\tProvides various third party annotations that have been computed for PDB chains.<br/>\\n\\tGet the third party features (a DAS - <i>FEATURES</i> request): <a href=\"/pdb/rest/das/pdbchainfeatures/features?segment=5pti.A\">/pdb/rest/das/pdbchainfeatures/features?segment=5pti.A</a><br/>\\n\\tGet the ATOM sequence of the PDB chain (a DAS - <i>SEQUENCE</i> request): <a href=\"/pdb/rest/das/pdbchainfeatures/sequence?segment=5pti.A\">/pdb/rest/das/pdbchainfeatures/sequence?segment=5pti.A</a><br/>\\n</li>\\n<li id=\"dasalignment\">\\n\\t<p><b>pdb_uniprot_mapping:</b></p> \\n\\tProvides a DAS source that is serving the alignment between UniProtKB and PDB, derived from the <a href=\"http://www.ebi.ac.uk/msd/sifts/\">SIFTS mappings</a>.<br/>\\t\\n\\tExample: Get all alignments for a PDB (a DAS - <i>ALIGNMENT</i> request) <a href=\"/pdb/rest/das/pdb_uniprot_mapping/alignment?query=4hhb\">/pdb/rest/das/pdb_uniprot_mapping/alignment?query=4hhb</a><br/>\\t\\n\\tExample: Get all alignments for a single PDB chain (a DAS - <i>ALIGNMENT</i> request) <a href=\"/pdb/rest/das/pdb_uniprot_mapping/alignment?query=4hhb.A\">/pdb/rest/das/pdb_uniprot_mapping/alignment?query=4hhb.A</a><br/>\\n</li>\\n\\n</ul>\\n\\n<p>For more documentation on the DAS protocol, see <a href=\"http://www.dasregistry.org/\">http://www.dasregistry.org/</a></p>\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"smiles\"></a>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Chemical Structure searches with SMILES or SMARTS strings</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n\\nThis query performs chemical structure searches using <a href=\"http://en.wikipedia.org/wiki/Simplified_molecular_input_line_entry_specification\">SMILES</a> and \\n<a href=\"http://en.wikipedia.org/wiki/Smiles_arbitrary_target_specification\">SMARTS</a> linear notations.\\n</br>\\nSearch types include <a href=\"/pdb/staticHelp.do?p=help/advancedsearch/chemSmiles.html\">exact, substructure, superstructure and similarity</a>.\\n\\n<ul>\\n\\n\\n<li>Exact:  find an exact structure match<br/>\\n</li>\\n<a href=\"/pdb/rest/smilesQuery?smiles=NC(=O)C1=CC=CC=C1&amp;search_type=exact\">/pdb/rest/smilesQuery?smiles=NC(=O)C1=CC=CC=C1&amp;search_type=exact</a>\\n\\n\\n<li>Substructure:  find ligands that contain the specified structure as a substructure<br/>\\n</li>\\n<a href=\"/pdb/rest/smilesQuery?smiles=NC(=O)C1=CC=CC=C1&amp;search_type=substructure\">/pdb/rest/smilesQuery?smiles=NC(=O)C1=CC=CC=C1&amp;search_type=substructure</a><br/>\\n\\n\\n<li>Superstructure: find ligands that are substructures (fragments) of the specified structure</br>\\n</li>\\n<a href=\"/pdb/rest/smilesQuery?smiles=OC(=O)c1ccc(OCc2ccccc2)cc1&amp;search_type=superstructure\">/pdb/rest/smilesQuery?smiles=OC(=O)c1ccc(OCc2ccccc2)cc1&amp;search_type=superstructure</a><br/>\\n\\n\\n<li>Similarity:  find structures that bind similar ligands.<br/>\\nSpecify a similarity threshold to change the degree of similarity in the [0...1] range:  0 dissimilar ... 1 identical.<br/>\\nThe similarity is based on the number of chemical features in common between the query and the target molecule.\\nSimilarity is calculated using <a href=\"http://en.wikipedia.org/wiki/Tanimoto_coefficient\">the Tanimoto Coefficient</a>.\\n</li>\\n<a href=\"/pdb/rest/smilesQuery?smiles=OC(=O)c1ccc(OCc2ccccc2)cc1&amp;search_type=similarity&amp;similarity=0.7\">/pdb/rest/smilesQuery?smiles=OC(=O)c1ccc(OCc2ccccc2)cc1&amp;search_type=similarity&amp;similarity=0.7</a>\\n\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n\\n<a name=\"search\"></a>\\n\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>SEARCH service</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"w_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n<p>\\nThis interface exposes the RCSB PDB <a href=\"/pdb/search/advSearch.do\">advanced search</a> interface as an XML Web Service.\\n</p>\\n\\n<p>\\nTo use this service, POST a XML representation of an advanced search to <a href=\"/pdb/rest/search\">/pdb/rest/search</a>. \\n</p>\\n\\n<h4>XML representation of advanced search</h4>\\n\\n<p>\\nEvery <a href=\"/pdb/search/advSearch.do\">advanced search</a> can be represented by XML. \\nTo view an example representation, simply execute an advanced search query and then click on the <a href=\"/pdb/results/results.do\">Result tab</a>. One of the links \\non the top of the page is <b>Query Details</b>.\\nClicking on this link will provide the XML representation of the query. \\n</p>\\n\\n<p> Every query is described by two data items:</p>\\n<ul>\\n<li><b>queryType:</b> the name of the class that is implementing the query</li>\\n<li><b>arguments:</b> depending on the type of query that is being executed one or more differently named arguments need to be provided.</li>\\n</ul>\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"w_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\t\\n\\n<script type=\"text/javascript\">\\nfunction advFunction(searchOption)\\n{\\n\\tvar queryToXML = searchOption.options[searchOption.selectedIndex].value;\\n\\n\\tif (queryToXML != \"default\") {\\n\\n    \\t$.post(\"/pdb/rest/xml/\",\\n    \\t    queryToXML,\\n    \\t    function (data){\\t  \\n\\t\\t\\t\\tvar obj = document.getElementById(\"xmlText\");\\n\\t\\t\\t\\tobj.value = data\\t    \\t\\t\\n    \\t    }\\n    \\t    );\\n\\t}\\n\\telse \\n\\t{\\n\\t\\talert(\"Please enter a query in the text box.\");\\n\\t}\\n    return false;\\n}\\n</script>\\n<h4>Sample XML Queries:</h4>\\n<select name=\"advSearch\" onChange=\"advFunction(this)\">\\n<option value=\"default\">Please Select...</option>\\n<optgroup label=\\'Quick Search\\' value=\\'Quick Search\\'>Quick Search</optgroup><option value=\\'HoldingsQuery\\'>All/Experimental Type/Molecule Type</option><optgroup label=\\'ID(s) and Keywords\\' value=\\'ID(s) and Keywords\\'>ID(s) and Keywords</optgroup><option value=\\'StructureIdQuery\\'>PDB ID(s)</option><option value=\\'EntityIdQuery\\'>Entity ID(s)</option><option value=\\'ChainIdQuery\\'>Chain ID(s)</option><option value=\\'PubmedIdQuery\\'>PubMed ID(s)</option><option value=\\'UpAccessionIdQuery\\'>UniProtKB Accession Number(s)</option><option value=\\'AdvancedKeywordQuery\\'>Text Search</option><option value=\\'TokenKeywordQuery\\'>mmCIF Keyword Search (Classification)</option><option value=\\'PfamIdQuery\\'>Pfam Accession Number(s)</option><option value=\\'UniprotGeneNameQuery\\'>UniProt Gene Name</option><option value=\\'SequenceClusterQuery\\'>Sequence Cluster Name</option><optgroup label=\\'Structure Annotation\\' value=\\'Structure Annotation\\'>Structure Annotation</optgroup><option value=\\'StructTitleQuery\\'>Structure Title</option><option value=\\'StructDescQuery\\'>Structure Description</option><option value=\\'MoleculeNameQuery\\'>Macromolecule Name</option><option value=\\'LargeStructureQuery\\'>Large Structures</option><optgroup label=\\'Deposition\\' value=\\'Deposition\\'>Deposition</optgroup><option value=\\'AdvancedAuthorQuery\\'>Author Name</option><option value=\\'DepositDateQuery\\'>Deposit Date</option><option value=\\'ReleaseDateQuery\\'>Release Date</option><option value=\\'ReviseDateQuery\\'>Revision Date</option><option value=\\'LastLoadQuery\\'>Latest Released Structures</option><option value=\\'ModifiedStructuresQuery\\'>Latest Modified Structures</option><option value=\\'SGProjectQuery\\'>Structural Genomics Project</option><optgroup label=\\'Structure Features\\' value=\\'Structure Features\\'>Structure Features</optgroup><option value=\\'ChainTypeQuery\\'>Macromolecule Type</option><option value=\\'NumberOfChainsQuery\\'>Number of Chains (Asymmetric Unit)</option><option value=\\'BiolUnitQuery\\'>Number of Chains (Biological Assembly)</option><option value=\\'NumberOfEntitiesQuery\\'>Number of Entities</option><option value=\\'StoichiometryQuery\\'>Protein Stoichiometry</option><option value=\\'PointGroupQuery\\'>Protein Symmetry</option><option value=\\'TreeQuery_20\\'>Protein Symmetry Browser</option><option value=\\'ModelCountQuery\\'>Number of Models</option><option value=\\'CloseContactsQuery\\'>Number of Disulfide Bonds</option><option value=\\'LinkConnectionQuery\\'>Link records</option><option value=\\'MolecularWeightQuery\\'>Molecular Weight (Structure)</option><option value=\\'SecondaryStructureQuery\\'>Secondary Structure Content</option><option value=\\'SecondaryStructureLengthQuery\\'>Secondary Structure Length</option><option value=\\'TreeQuery_11\\'>SCOP Classification Browser</option><option value=\\'TreeQuery_12\\'>CATH Classification Browser</option><option value=\\'TreeQueryExpression_1\\'>Taxonomy Browser</option><optgroup label=\\'Sequence Features\\' value=\\'Sequence Features\\'>Sequence Features</optgroup><option value=\\'SequenceQuery\\'>Sequence (BLAST/FASTA/PSI-BLAST)</option><option value=\\'WildTypeProteinQuery\\'>Wild Type Protein</option><option value=\\'BlastXQuery\\'>Translated Nucleotide Sequence (BLASTX)</option><option value=\\'MotifQuery\\'>Sequence Motif</option><option value=\\'SequenceLengthQuery\\'>Chain Length</option><option value=\\'TreeEntityQuery_13\\'>Genome Location Browser</option><optgroup label=\\'Chemical Components\\' value=\\'Chemical Components\\'>Chemical Components</optgroup><option value=\\'ChemCompNameQuery\\'>Chemical Name</option><option value=\\'ChemCompIdQuery\\'>Chemical ID(s)</option><option value=\\'ChemCompDescriptorQuery\\'>InChI Descriptor</option><option value=\\'ChemSmilesQuery\\'>Chemical structure (SMILES)</option><option value=\\'ChemFormulaWeightQuery\\'>Molecular Weight (Chemical component)</option><option value=\\'ChemCompFormulaQuery\\'>Chemical Formula</option><option value=\\'ChemCompTypeQuery\\'>Chemical Component Type</option><option value=\\'BindingAffinityQuery\\'>Binding Affinity</option><option value=\\'NoLigandQuery\\'>Has Ligand(s)</option><option value=\\'NoModResQuery\\'>Has Modified Residue(s)</option><option value=\\'ChemCompSubCompQuery\\'>Sub-components</option><optgroup label=\\'Biologically Interesting Molecules\\' value=\\'Biologically Interesting Molecules\\'>Biologically Interesting Molecules</optgroup><option value=\\'BirdQuery\\'>Biologically Interesting Molecules (from BIRD)</option><optgroup label=\\'Biology\\' value=\\'Biology\\'>Biology</optgroup><option value=\\'TreeEntityQuery_1\\'>Source Organism Browser (NCBI)</option><option value=\\'ExpressionOrganismQuery\\'>Expression Organism</option><option value=\\'TreeEntityQuery_3\\'>Enzyme Classification Browser</option><option value=\\'EnzymeClassificationQuery\\'>Enzyme Classification</option><option value=\\'TreeEntityQuery_4\\'>Biological Process Browser (GO)</option><option value=\\'TreeEntityQuery_5\\'>Cell Component Browser (GO)</option><option value=\\'TreeEntityQuery_6\\'>Molecular Function Browser (GO)</option><option value=\\'TreeQuery_16\\'>Transporter Classification Browser</option><optgroup label=\\'Methods\\' value=\\'Methods\\'>Methods</optgroup><option value=\\'ExpTypeQuery\\'>Experimental Method</option><option value=\\'ResolutionQuery\\'>X-ray Resolution</option><option value=\\'AverageBFactorQuery\\'>X-ray Average B Factor</option><option value=\\'XrayRefinementQuery\\'>Refinement R Factors</option><option value=\\'XrayDiffrnSourceQuery\\'>Diffraction Source</option><option value=\\'MethodToDetermineStructQuery\\'>Structure Determination Method</option><option value=\\'XrayReflnsQuery\\'>Reflections</option><option value=\\'XrayCellQuery\\'>Cell Dimensions</option><option value=\\'SoftwareQuery\\'>Software</option><option value=\\'SpaceGroupQuery\\'>Space Group</option><option value=\\'CrystalQuery\\'>Crystal Properties</option><option value=\\'EmAssemblyQuery\\'>EM Assembly</option><option value=\\'DetectorTypeQuery\\'>Detector</option><optgroup label=\\'Publication\\' value=\\'Publication\\'>Publication</optgroup><option value=\\'CitationTwoQuery\\'>Citation</option><option value=\\'TreeQuery_15\\'>Medical Subject Headings Browser</option><option value=\\'PubMedQuery\\'>PubMed Abstract</option><optgroup label=\\'Misc\\' value=\\'Misc\\'>Misc</optgroup><option value=\\'ExternalLinkQuery\\'>Has External Links</option>\\n</select>\\n<p>\\nThe Query is displayed in the Textbox below\\n</p>\\n\\n\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Example:</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"db_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n\\n\\n\\n<pre>\\n&lt;orgPdbQuery&gt;    \\n    &lt;queryType&gt;org.pdb.query.simple.UpAccessionIdQuery&lt;/queryType&gt;\\n    &lt;description&gt;Simple query for a list of UniprotKB Accession IDs: P50225&lt;/description&gt;   \\n    &lt;accessionIdList&gt;P50225&lt;/accessionIdList&gt;\\n&lt;/orgPdbQuery&gt;\\n</pre>\\n\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"db_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n<p>Now let\\'s <b>POST</b> this query against the Web Service:</p>\\n<script type=\"text/javascript\">\\n\\nfunction submitForm(){\\n\\n\\tvar sorting = document.getElementById(\\'sortField\\').value;\\n\\tvar queryXML = document.getElementById(\\'xmlText\\').value;\\n\\n\\tif (queryXML.length > 0) {\\n\\n\\t\\tqueryXML = checkComparator(queryXML);\\n\\t\\n\\t\\tvar xmlText = encodeURIComponent(queryXML);\\n\\n\\t\\tvar restUrl=\"/pdb/rest/search/\"+(sorting?\"?sortfield=\"+sorting:\"\")\\n\\t    document.getElementById(\\'restUrl\\').innerHTML=\"Posted XML to: \"+restUrl;\\n\\t\\t\\n    \\t$.post(\"/pdb/rest/search/?req=browser\"+(sorting?\"&sortfield=\"+sorting:\"\"),\\n    \\t    xmlText,\\n    \\t    function (data){\\t  \\n\\t    \\t\\t\\n\\t    \\t\\tvar spl = data.split(\"\\\\n\");\\n\\n\\t    \\t\\tvar rs=0;\\n\\t    \\t\\tvar h = \"<p>\";\\n\\t    \\t\\tvar qrId=null;\\n\\t    \\t\\tfor (var spliti = 0; spliti < spl.length; spliti++){\\n\\t    \\t\\t\\tvar it = spl[spliti].replace(/[\\\\n\\\\t\\\\r]/g,\"\");\\n\\t\\t    \\t\\tif (it.length==4) {\\n\\t\\t    \\t\\t\\th+=\\'<a href=\"/pdb/explore.do?structureId=\\'+it+\\'\">\\'+it+\\'</a> \\';\\n\\t\\t    \\t\\t\\trs++;\\n\\t\\t    \\t\\t} else if ((it.length>4)&&(it.indexOf(\".\")>0))\\n\\t\\t\\t\\t\\t{\\n\\t\\t\\t\\t\\t\\th+=\\'<a href=\"/pdb/explore/remediatedSequence.do?structureId=\\'+it.substring(0,4)+\\'&params.chainEntityStrategyStr=all&forcePageForChain=\\'+it.substring(5)+\\'\">\\'+it+\\'</a> \\';\\n\\t\\t\\t\\t\\t\\trs++;\\n\\t\\t\\t\\t\\t} else if ((it.length>4)&&(it.indexOf(\":\")>0))\\n\\t\\t\\t\\t\\t{\\n\\t\\t\\t\\t\\t\\th+=\\'<a href=\"/pdb/explore/remediatedSequence.do?structureId=\\'+it.substring(0,4)+\\'\">\\'+it+\\'</a> \\';\\n\\t\\t\\t\\t\\t\\trs++;\\n\\t\\t\\t\\t\\t}\\n\\t\\t\\t\\t\\telse if(it.length>4 && spliti>0 && rs>0) {\\n\\t\\t    \\t\\t\\th+=\\'<br/><br/><a href=\"/pdb/results/results.do?qrid=\\'+it+\\'\">View results</a> \\';\\n\\t\\t    \\t\\t}\\n\\t    \\t\\t}\\t    \\t\\t\\n\\t    \\t\\th+=\"</p>\";\\n\\t    \\t\\th=\"<p>\"+rs+\" results</p>\"+h;\\n\\t    \\t\\t\\t    \\t\\t\\n\\t    \\t\\t$(\\'#restResults\\').html(h);\\n    \\t    }\\n    \\t    );\\n\\t}\\n\\telse \\n\\t{\\n\\t\\talert(\"Please enter a query in the text box.\");\\n\\t}\\n    return false;\\n\\t\\t\\n}\\n\\nfunction checkComparator(queryXML) {\\n\\n\\twhile (queryXML.indexOf(\".comparator=< \")>-1) {\\n\\t   \\tqueryXML = queryXML.replace(\".comparator=< \", \".comparator=<![CDATA[<]]>\");\\n\\t}\\n\\t\\n\\twhile (queryXML.indexOf(\".comparator><<\")>-1) {\\n\\t\\tqueryXML = queryXML.replace(\".comparator><<\", \".comparator><![CDATA[<]]><\");\\n\\t}\\n\\t\\n\\twhile (queryXML.indexOf(\".comparator=> \")>-1) {\\n\\t   \\tqueryXML = queryXML.replace(\".comparator=> \", \".comparator=<![CDATA[>]]>\");\\n\\t}\\t\\n\\n\\twhile (queryXML.indexOf(\".comparator>><\")>-1) {\\n\\t\\tqueryXML = queryXML.replace(\".comparator>><\", \".comparator><![CDATA[>]]><\");\\n\\t}\\n\\treturn queryXML;\\n\\t\\n}\\n</script>\\n\\n\\n<textarea id=\"xmlText\" rows=\"10\" class=\"perc100\">\\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n\\n<orgPdbQuery>    \\n\\n    <queryType>org.pdb.query.simple.UpAccessionIdQuery</queryType>\\n\\n    <description>Simple query for a list of UniprotKB Accession IDs: P50225</description>   \\n\\n    <accessionIdList>P50225</accessionIdList>\\n\\n</orgPdbQuery>\\n\\n</textarea>\\n<br/>\\nSort results by:\\n<select name=\"sortField\" id=\"sortField\">\\n<option value=\"\">&#x21E7; PDB ID (default)</option>\\n<option value=\"Release Date\">&#x21E7; Release Date</option>\\n<option value=\"Residue Count\">&#x21E7; Residue Count</option>\\n<option value=\"Resolution\">&#x21E7; Resolution</option>\\n<option value=\"rank Descending\">&#x21E9; Relevance/E value</option>\\n</select>\\n&nbsp;&nbsp;&nbsp;\\n<input type=\"submit\" value=\"POST XML\" onclick=\"submitForm();\" title=\"The results  will appear in the box below.\" class=\"tooltip\"/>\\n\\n<input type=\"button\" value=\"RESET\" onclick=\"document.getElementById(\\'xmlText\\').value=\\'\\'\" title=\"Clear the query above.\" class=\"tooltip\" />\\n<div id=\"restUrl\">\\n</div>\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Results will go here:</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"lb_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n<div id=\"restResults\">\\n</div>\\n\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"lb_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n<input type=\"button\" value=\"CLEAR RESULTS\" onclick=\"document.getElementById(\\'restResults\\').innerHTML=\\'\\';window.location.reload();\" title=\"Clear the results above.\" class=\"tooltip\">\\t\\n\\t\\n\\n\\n\\n\\n\\n\\n\\t\\n\\t<div>\\n\\t\\n\\t<div class=\"db_head3\">\\n\\t\\t\\n\\t\\t\\n\\t\\t<div class=\"box_header\">\\n\\n\\t\\t\\t<b>Examples:</b>\\n\\n\\t\\t</div>\\n\\n\\n\\n\\t</div>\\n\\t\\n\\n\\n\\t<div class=\"db_body3\">\\n\\n\\t\\t\\n\\t\\t\\t\\n\\t\\t\\t<div class=\"box_contentWrapper\">\\n\\n\\t\\t\\n\\n\\n\\n<ul>\\n\\t<li>View a Perl example that submits a simple search for organism:\\n\\t\\t<a href=\"static.do?p=/software/webservices/search_organism.jsp\">search_organism.pl</a>\\n\\t</li>\\n\\t<li>View a Java example that submits a more complex Advanced Search doing a BLAST search and only looking at X-RAY structures.\\n\\t\\t<a href=\"static.do?p=/software/webservices/PostXMLQuery.jsp\">PostXMLQuery.java</a>\\n\\t</li>\\n\\t<li>View a Python example that submits a simple search for all Solid State NMR proteins:\\n\\t\\t<a href=\"static.do?p=/software/webservices/search_nmr.jsp\">search_nmr.py</a>\\n\\t</li>\\n</ul>\\n\\n\\n\\n\\n\\t\\t\\t\\n\\n\\t\\t</div>\\n\\t\\n\\t</div>\\n\\n</div>\\n\\n\\n\\t<div class=\"db_foot3\">\\n\\t\\t\\t\\t\\n\\t\\t&nbsp;\\n\\t\\t\\t\\n\\t</div>\\n\\n\\n\\n\\n<div class=\"se_spacer\">&nbsp;</div>\\n\\n<hr/>\\n\\n\\n\\n  \\n                            </div>\\n                        </td>\\n                    </tr>\\n                </table>\\n            </div>\\n            <!-- NEW FOOTER -->\\n<div id=\"footer_main\">\\n    <div class=\"footercontainer\">\\n        <div class=\"col-md-8\">\\n            <ul>\\n                <li><span class=\"row_header\">About</span></li>\\n                <li><a href=\"/pdb/static.do?p=general_information/about_pdb/index.html\">About Us</a></li>\\n                <li><a href=\"/pdb/static.do?p=general_information/about_pdb/policies_references.html#References\">Citing Us</a></li>\\n                <li><a href=\"/pdb/static.do?p=general_information/news_publications/index.html\">Publications</a></li>\\n                <li><a href=\"/pdb/static.do?p=general_information/about_pdb/contact/pdb_members.html\">Team</a></li>\\n                <li><a href=\"/pdb/static.do?p=general_information/about_pdb/contact/job_listings.html\">Careers</a></li>\\n                <li><a href=\"/pdb/static.do?p=general_information/about_pdb/policies_references.html\">Usage & Privacy</a></li>\\n            </ul>\\n            <ul>\\n                <li><span class=\"row_header\">Help</span></li>\\n                <li><a href=\"/pdb/home/contactUs.do\">Contact Us</a></li>\\n                <li><a href=\"/pdb/static.do?p=help/index.html\">Help Topics</a></li>\\n                <li><a href=\"/pdb/static.do?p=home/faq.html\">Website FAQ</a></li>\\n                <li><a href=\"/pdb/browser/browsercheck.do\">Browser Test</a></li>\\n                <li><a href=\"/pdb/static.do?p=help/glossary.html\">Glossary</a></li>\\n            </ul>\\n            <ul>\\n                <li><span class=\"row_header\">RCSB Partners</span></li>\\n                <li><a href=\"http://ndbserver.rutgers.edu/\" target=\"_blank\">Nucleic Acid Database</a></li>\\n                <li><a href=\"http://sbkb.org/\" target=\"_blank\">Structural Biology Knowledgebase</a></li>\\n            </ul>\\n        </div>\\n        <div class=\"col-md-4\">\\n            <p>The RCSB PDB (<a href=\"http://nar.oxfordjournals.org/content/28/1/235.abstract\" target=\"blank\">citation</a>) is managed by two members of the <br>Research Collaboratory for Structural Bioinformatics: <br>Rutgers and UCSD/SDSC\\n            </p>\\n\\n            <img src=\"/pdb/v2/common/images//Logo_Rutgers-UCSD-SDSC.png\" alt=\"Rutgers UCSD SDSC\" usemap=\"#university-map\" width=\"331\" height=\"26\">\\n            <map name=\"university-map\" id=\"university-map\">\\n                <area shape=\"rect\" coords=\"0,0,115,26\" href=\"http://www.rutgers.edu/\" alt=\"Rutgers\" target=\"_blank\">\\n                <area shape=\"rect\" coords=\"116,0,225,26\" href=\"http://ucsd.edu/\" alt=\"UCSD\" target=\"_blank\">\\n                <area shape=\"rect\" coords=\"226,0,331,26\" href=\"http://www.sdsc.edu/\" alt=\"SDSC\" target=\"_blank\">\\n            </map>\\n            <div class=\"clear\"></div>\\n\\n            <p>RCSB PDB is a member of the <a href=\"http://www.wwpdb.org/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_wwpdb.png\" alt=\"wwPDB\" class=\"wwpdblogo\"></a>\\n                <a href=\"http://www.emdatabank.org/\" target=\"_blank\"><img src=\"/pdb/v2/common/images/Logo_EMDatabank.png\" alt=\"EMDataBank\" id=\"emdatabanklogo\"></a>\\n            </p>\\n        </div>\\n    </div>\\n    <div class=\"clear\"></div>\\n</div>\\n<div id=\"footer_grant\">\\n    <div class=\"footercontainer\">\\n        <p>The RCSB PDB is funded by a grant (DBI-1338415) from the <a href=\"http://www.nsf.gov/\" target=\"_blank\">National Science Foundation</a>, the <a href=\"http://www.nih.gov/\" target=\"_blank\">National Institutes of Health</a>, and the <a href=\"http://science.energy.gov/\" target=\"_blank\">US Department of Energy</a>.</p>\\n    </div>\\n</div>\\n\\n\\n\\n<div style=\"text-align: center; height: 30px; width: 90px; position: fixed; bottom: 0%; right: 1%; z-index: 999;\" class=\"hidden-print\">\\n    <button class=\"btn btn-default hidden-xs\" id=\"feedbackLink-Footer\" style=\"border-color: rgb(94, 148, 195); border-width: 2px;\">Contact Us</button>\\n</div>\\n<script type=\"text/javascript\" src=\"/pdb/general_information/about_pdb/contact/Jira-ContactUs-HelpDesk.js\"></script>\\n\\n\\n<script type=\"text/javascript\">window.ATL_JQ_PAGE_PROPS = {\\n    \"triggerFunction\": function (showCollectorDialog) {\\n        //Requries that jQuery is available!\\n        if ($(\\'#feedbackLink-ContactUs\\').length > 0) {\\n            // We have TWO buttons to attach listeners too\\n            document.getElementById(\"feedbackLink-ContactUs\").addEventListener(\"click\", function (e) {\\n                e.preventDefault();\\n                showCollectorDialog();\\n                return false\\n            }, false);\\n            document.getElementById(\"feedbackLink-Footer\").addEventListener(\"click\", function (e) {\\n                e.preventDefault();\\n                showCollectorDialog();\\n                return false\\n            }, false);\\n        } else {\\n            // Only the footer link exists - so listen to it\\n            document.getElementById(\"feedbackLink-Footer\").addEventListener(\"click\", function (e) {\\n                e.preventDefault();\\n                showCollectorDialog();\\n                return false\\n            }, false);\\n        }\\n    }};\\n</script>\\n\\n        </td>\\n    </tr>\\n</table>\\n<!-- What is this? -->\\n\\n\\n\\n\\n<script type=\"text/javascript\">\\n    var gaJsHost = ((\"https:\" == document.location.protocol) ? \"https://\" : \"http://\");\\n    document.write(unescape(\"%3Cscript src=\\'\" + gaJsHost + \"stats.g.doubleclick.net/dc.js\\' type=\\'text/javascript\\'%3E%3C/script%3E\"));\\n</script>\\n<script type=\"text/javascript\">\\n    if (typeof _gat != \\'undefined\\') {\\n        var pageTracker = _gat._getTracker(\"UA-3923365-3\");\\n        pageTracker._initData();\\n        pageTracker._trackPageview();\\n        _gat._anonymizeIp();\\n    } else {\\n        // Create a dummy Google Analytics script that prevents JavaScript error messages.\\n        var pageTracker = {\\n            _addIgnoredOrganic: function (newIgnoredOrganicKeyword) {},\\n            _addIgnoredRef: function (newIgnoredReferrer) {},\\n            _addItem: function (orderId, sku, name, category, price, quantity) {},\\n            _addOrganic: function (newOrganicEngine, newOrganicKeyword, opt_prepend) {},\\n            _addTrans: function (orderId, affiliation, total, tax, shipping, city, state, country) {},\\n            _clearIgnoredOrganic: function () {},\\n            _clearIgnoredRef: function () {},\\n            _clearOrganic: function () {},\\n            _cookiePathCopy: function (newPath) {},\\n            _deleteCustomVar: function (index) {},\\n            _getName: function () {},\\n            _setAccount: function () {},\\n            _getAccount: function () {},\\n            _getClientInfo: function (a) {},\\n            _getDetectFlash: function (a) {},\\n            _getDetectTitle: function (a) {},\\n            _getLinkerUrl: function (targetUrl, useHash) {},\\n            _getLocalGifPath: function () {},\\n            _getServiceMode: function () {},\\n            _getVersion: function () {},\\n            _getVisitorCustomVar: function (index) {},\\n            _link: function (targetUrl, useHash) {},\\n            _linkByPost: function (formObject, useHash) {},\\n            _setAllowAnchor: function (bool) {},\\n            _setAllowHash: function (bool) {},\\n            _setAllowLinker: function (bool) {},\\n            _setCampContentKey: function (newCampContentKey) {},\\n            _setCampMediumKey: function (newCampMedKey) {},\\n            _setCampNameKey: function (newCampNameKey) {},\\n            _setCampNOKey: function (newCampNOKey) {},\\n            _setCampSourceKey: function (newCampSrcKey) {},\\n            _setCampTermKey: function (newCampTermKey) {},\\n            _setCampaignCookieTimeout: function (cookieTimeoutMillis) {},\\n            _setCampaignTrack: function (bool) {},\\n            _setClientInfo: function (bool) {},\\n            _setCookiePath: function (newCookiePath) {},\\n            _setCustomVar: function (index, name, value, opt_scope) {},\\n            _setDetectFlash: function (bool) {},\\n            _setDetectTitle: function (bool) {},\\n            _setDomainName: function (newDomainName) {},\\n            _setLocalGifPath: function (newLocalGifPath) {},\\n            _setLocalRemoteServerMode: function () {},\\n            _setLocalServerMode: function () {},\\n            _setReferrerOverride: function (newReferrerUrl) {},\\n            _setRemoteServerMode: function () {},\\n            _setSampleRate: function (newRate) {},\\n            _setSessionCookieTimeout: function (cookieTimeoutMillis) {},\\n            _setVisitorCookieTimeout: function (cookieTimeoutMillis) {},\\n            _trackEvent: function (category, action, opt_label, opt_value) {},\\n            _trackPageview: function (opt_pageURL) {},\\n            _trackTrans: function () {}\\n        };\\n    }\\n</script>\\n\\n\\n<script>\\n    $(\"a[href$=\\'pdf\\']\").each(function (index) {\\n        pdfLabel = $(this).attr(\\'href\\');\\n        pdfOnClick = \"console.log(\\'pdfLabel\\'); var that=this;pageTracker._trackEvent(\\'PDF\\', \\'Download\\', \\'\" + pdfLabel + \"\\');setTimeout(function(){location.href=that.href;},200);return false;\";\\n        $(this).attr(\"onClick\", pdfOnClick);\\n    });\\n</script>\\n</body>\\n</html>\\n\\n'"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
